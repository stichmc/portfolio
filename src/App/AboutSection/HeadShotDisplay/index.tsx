import { div as MotionDiv } from "motion/react-client";

export interface HeadShotDisplayProps {
  headShotImage: string;
}

const HeadShotDisplay = ({ headShotImage }: HeadShotDisplayProps) => {
  return (
    <div className="relative w-[400px] h-[400px]">
      <div className="absolute inset-0 z-0 flex items-center justify-center">
        <MotionDiv
          animate={{
            scale: [1, 2, 2, 1, 1],
            rotate: [0, 0, 180, 180, 0],
            borderRadius: ["0%", "0%", "50%", "50%", "0%"],
          }}
          transition={{
            duration: 8,
            ease: "easeInOut",
            times: [0, 0.2, 0.5, 0.8, 1],
            repeat: Infinity,
            repeatDelay: 2,
            delay: 2,
          }}
          className="bg-card-foreground w-[200px] h-[200px]"
        />
      </div>
      <div className="relative z-10 flex items-center justify-center rounded-full w-full h-full overflow-hidden border-card-foreground border-4">
        <img className="w-full h-full object-cover" src={headShotImage} alt="michael-headshot" />
      </div>
    </div>
  );
};

export default HeadShotDisplay;
