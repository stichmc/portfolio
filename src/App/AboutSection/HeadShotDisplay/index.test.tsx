import { expect, test, beforeEach } from "vitest";
import { render } from "@testing-library/react";
import HeadShotDisplay, { HeadShotDisplayProps } from ".";

let props: HeadShotDisplayProps;

beforeEach(() => {
  props = {
    headShotImage: "link-to-image",
  };
});

test("HeadShotDisplay", () => {
  const rendering = render(<HeadShotDisplay {...props} />);
  expect(rendering).toMatchSnapshot();
});
