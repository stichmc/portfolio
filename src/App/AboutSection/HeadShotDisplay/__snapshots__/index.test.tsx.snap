// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HeadShotDisplay 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="relative w-[400px] h-[400px]"
      >
        <div
          class="absolute inset-0 z-0 flex items-center justify-center"
        >
          <div
            class="bg-card-foreground w-[200px] h-[200px]"
          />
        </div>
        <div
          class="relative z-10 flex items-center justify-center rounded-full w-full h-full overflow-hidden border-card-foreground border-4"
        >
          <img
            alt="michael-headshot"
            class="w-full h-full object-cover"
            src="link-to-image"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="relative w-[400px] h-[400px]"
    >
      <div
        class="absolute inset-0 z-0 flex items-center justify-center"
      >
        <div
          class="bg-card-foreground w-[200px] h-[200px]"
        />
      </div>
      <div
        class="relative z-10 flex items-center justify-center rounded-full w-full h-full overflow-hidden border-card-foreground border-4"
      >
        <img
          alt="michael-headshot"
          class="w-full h-full object-cover"
          src="link-to-image"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
